ansible/deploy.yml

```diff
@@ -0,0 +1,48 @@
+---
+- hosts: hypervisors
+  tasks:
+    - include_tasks: host_setup.yml
+
+- hosts: controllers
+  tasks:
+    - include_tasks: schedule.yml
```

Comment:
Should the scheduling ultimately take account of existing allocations?

Reply:
That's a good question. I suppose it depends on whether we want Tenks to support multiple different 'cluster deployments'. How I currently see it, that would work by calling Tenks *n* times with *n* different `allocations.yml` file paths, but that wouldn't take into account allocations made by 'cluster deployments' of other Tenks runs.

I was imagining that `schedule.py` would eventually query system stats (e.g. memory/CPU usage, number of VMs currently running under Libvirt, etc.) when deciding where to schedule VMs, so perhaps doing that would indirectly allow the existence of VMs from other Tenks runs to influence placement for this run.

Reply:
Interesting point about multiple tenks clusters, but I was mostly thinking about rerunning tenks with an existing allocations.yml to e.g. add a VM.

Reply:
Ah, true. We could pass the existing contents (if any) of `allocations.yml` into `schedule.py`, which will tell us how many VMs are current scheduled on each host.
Then,
```
n=len(new_vms) - len(current_vms)

case n of:
```
  `== 0` :No action required; no-op
  `> 0`: Schedule *n* VMs onto our hypervisors as normal
  `< 0`: Delete *-n* VMs - perhaps use similar scheduling heuristics to decide which hypervisors have the heaviest current load.